frontend/ui/widget/inputdialog.lua

```diff
@@ -416,30 +417,79 @@ function InputDialog:init()
         end
     end
 
+    -- checkbox
+    if self.check_button_text then
+        self.check_button = self.check_button or CheckButton:new{
+            text = self.check_button_text,
+            face = Font:getFace("smallinfofont"),
+            checked = self.check_button_checked,
+            callback = function()
+                if not self.check_button.checked then
+                    self.check_button:check()
+                else
+                    self.check_button:unCheck()
+                end
+                self:onShow()
+            end,
+            padding = self.padding,
+            margin = self.margin,
+            bordersize = self.bordersize,
+        }
+    end
+
     -- Final widget
-    self.dialog_frame = FrameContainer:new{
-        radius = self.fullscreen and 0 or Size.radius.window,
-        padding = 0,
-        margin = 0,
-        bordersize = self.border_size,
-        background = Blitbuffer.COLOR_WHITE,
-        VerticalGroup:new{
-            align = "left",
-            self.title_widget,
-            self.title_bar,
-            self.description_widget,
-            vspan_before_input_text,
-            CenterContainer:new{
-                dimen = Geom:new{
-                    w = self.width,
-                    h = self._input_widget:getSize().h,
+    if not self.check_button then -- without checkbutton
+        self.dialog_frame = FrameContainer:new{
```

Comment:
Not sure we should hack the already complex InputDialog to add it a checkbutton :)
Asking @hius07 who has become familiar with it: what do you think?
Should it be made a subclass (like we have multiinputdialog.lua)?
Or is it just ok and a generic enough feature? (although it might not work in all combinations, like with fullscreen=true, or does it?).

If this is fine, instead of duplicating the construction of `self.dialog_frame = FrameContainer:new{` (and the risk to forget to update it similarly in the 2 places), can the `vspan_before_input_text` + `self.check_button` just be inserted dynamically in a normal self.dialog_frame (if that's all the differences I see)?

Reply:
> Not sure we should hack the already complex InputDialog to add it a checkbutton

I like it, it's a good feature and can be used elsewhere in the future.

Propose to change `face = Font:getFace("smallinfofont")` in checkbutton.lua, to fix it everywhere, and remove it from here.

Reply:
Can it be useful to have several checkbuttons? E.g.
`[ ] use regexp`
`[ ] case sensitive`
`[ ] cycle search`
Like we can have more usual buttons.

Reply:
> Propose to change `face = Font:getFace("smallinfofont")` in checkbutton.lua, to fix it everywhere, and remove it from here.

Don't know if that breaks the layout elsewhere?

Reply:
> Can it be useful to have several checkbuttons? E.g.
> `[ ] use regexp`
> `[ ] case sensitive`
> `[ ] cycle search`
> Like we can have more usual buttons.

Thats no problem, if it is consensus to do that.

Reply:
I don't mind having more checkbox/radios in the Search dialog (as it's seen only when invoking it from the menu - I do more search by just selecting text).
What I do mind is extending InputDialog for specific needs with new options that works only with a subset of the existing options.
Might be saner to create a new widget that extends it, like openwithdialog.lua (either in a new widget file like InputDialogWithCheckboxes.lua, InputDialogWithRadiosButtons.lua - or just creating it in readersearch.lua).

Reply:
I see and understand your point. 
I will go for InputDialogWithCheckboxes.lua on the weekend.
Then I can add the ignoorecase option too. Are there any other checkboxes we want for search?

Reply:
> Are there any other checkboxes we want for search?

Just some idea, for fun :):
`[x] Match full words only` - incompatible with regexp as this would translate the word to the regexp equivalent to match word boundaries on each side of the input text (dunno if the regexp word boundaries work with all unicode (arabic, chinese...),)

Reply:
`[ ] Search from the beginning of a book`
instead of the current position, applicable for search forward

Use regexp, Case sensitive - all unchecked by default.

Shouldn't checkboxes be left aligned?